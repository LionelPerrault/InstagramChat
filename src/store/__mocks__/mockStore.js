import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
