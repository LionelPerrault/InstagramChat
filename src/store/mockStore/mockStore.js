import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
