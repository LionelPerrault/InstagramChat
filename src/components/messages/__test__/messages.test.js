import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../store/__mocks__/mockStore'
import Messages, { PureMessages } from '../messages'
import { shallow } from 'enzyme'

describe('Messages Component', () => {
  const mockFn = jest.fn()

  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <Messages />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should hide spinner when loading = false', () => {
    const wrapper = shallow(<PureMessages dispatch={mockFn} />)
    wrapper.setState({ loading: false })
    expect(wrapper.find('IsLoading').prop('loading')).toEqual(false)
  })

  it('should show <Conversation/> when showConversation == true', () => {
    const wrapper = shallow(<PureMessages dispatch={mockFn} />)
    wrapper.setState({
      selectedCon: {
        con_id: 44,
        unreadMssgs: 0,
      },
      showConversation: true,
    })
    expect(wrapper.find('Connect(Conversation)').exists()).toBe(true)
  })
})
