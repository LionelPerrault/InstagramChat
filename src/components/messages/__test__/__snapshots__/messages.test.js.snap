// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Component should match snapshot 1`] = `
<div>
  <div
    className="page_loading"
  >
    <div
      className="spinner"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="messages_senapati cLoading"
    style={
      Object {
        "animation": "bcCCNc 300ms ease 0s 1 normal none running",
        "backfaceVisibility": "visible",
      }
    }
  >
    <div
      className="mssg_left"
    >
      <div
        className="mssg_new"
      >
        <a
          className="pri_btn"
          href="#"
          onClick={[Function]}
        >
          <i
            className="fas fa-globe"
          />
          <span>
            Online users
          </span>
        </a>
        <a
          className="pri_btn"
          href="#"
          onClick={[Function]}
        >
          <i
            className="fas fa-plus"
          />
          <span>
            New conversation
          </span>
        </a>
      </div>
      <span
        className="con_count"
      >
        2 conversations
      </span>
      <div
        style={
          Object {
            "animation": "bcCCNc 300ms ease 0s 1 normal none running",
            "backfaceVisibility": "visible",
          }
        }
      >
        <div
          className="mssg_sr mt_24"
          onClick={[Function]}
        >
          <img
            src="/users/7/avatar.jpg"
          />
          <div
            className="m_sr_content"
          >
            <span
              className="m_sr_username"
            >
              ghalib
            </span>
            <span
              className="m_sr_light"
            >
              eo
            </span>
          </div>
          <span
            className="m_sr_time"
          >
            a month
          </span>
        </div>
        <div
          className="mssg_sr mt_40"
          onClick={[Function]}
        >
          <img
            src="/users/10/avatar.jpg"
          />
          <div
            className="m_sr_content"
          >
            <span
              className="m_sr_username"
            >
              noddy
            </span>
            <span
              className="m_sr_light"
            />
          </div>
          <span
            className="m_sr_time"
          />
        </div>
      </div>
    </div>
    <div
      className="mssg_right"
    >
      <div
        style={
          Object {
            "marginTop": 77,
          }
        }
      >
        <div
          className="home_last_mssg"
          style={
            Object {
              "border": "",
            }
          }
        >
          <img
            src="/images/large.jpg"
          />
          <span
            className="nothingMssg"
          >
            Please select a conversation
          </span>
          <span
            className="secondMssg"
          >
            
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
